### ISSUE TITLE
Troubleshooting Intel AMT

### ISSUE THREAD

**Issue by plwc** [Ylianst/MeshCentral#2888]
I've been running MC for a long time now very successfully. Everything is working perfectly for me except trying to configure AMT systems.

Right now I am running 0.8.77, I have allowed 4433 straight through to my MC server and I have confirmed that my router is passing through that traffic when I try to connect to that port on another application.

I have many AMT systems running completely unconfigured on vPro Intel NUCs and it shows up as "Not Activated (Pre), v11.8.65" with varying version numbers on the general tab for each device.

When I try any of the AMT options under the device group there is a flurry of activity showing on the firewall from the agents but nothing on 4433 and desite having all four AMT options ticked on the tracing page of the server, nothing shows up.

```
{
  "settings": {
    "Cert": "remote.mydomain.com",
    "Wanonly": true,
    "Port": 443,
    "AliasPort": 8888,
    "AgentPort": 444,
    "MpsPort": 4433,
    "Minify": false
  }
}
```

I'm think I must be doing something wrong but I can't figure out what.

**Comment by plwc** [comment]
I've tried setting up an AMT only group and using meshcmd to configure them. It mostly works except I get:

LME connection failed, HECI Connection Error [OVERLAPPED_ERROR: 31]

Everything else seems to go through okay and the last thing I get is:
Checking Intel AMT state...
Intel AMT connected.
Fetching hardware inventory.
Done.

The agents are added to the AMT only group but aren't showing as connected. I am able to use meshcmd with the password that was created to get some info but still get the LME connection failed.

Under the device details via the agent I get "Activated CCM, v11.8.65, No Credentials" and if I enter the correct password it just permanently shows Trying Credentials until I delete it.

If I type amtmanager in to the server console I get "No managed Intel AMT devices"

**Comment by Ylianst** [comment]
Hi. For Intel AMT only device groups, you may want to [check out this YouTube video on the topic](https://www.youtube.com/watch?v=vS9xRcgyqNI). I also have an [older video here](https://www.youtube.com/watch?v=mxkGRpHHvgA) that will be of use.

I think all you need to do is set and Intel AMT policy in the device group. In your case, I imagine you are running on the Internet and not on a local network so your server is in WAN mode. In that case, CIRA needs to be setup within Intel AMT for AMT to connect to MeshCentral. You need to setup a AMT policy with CIRA enabled and run "meshcmd amtconfig" again to get that working. The older video should help with that.

For the [OVERLAPPED_ERROR: 31], you can safely ignore that since you already have LMS running on the Intel AMT device. Bryan should be investigating this and cleaning that up so that does not show up anymore.

**Comment by plwc** [comment]
I have watched both of those videos multiple times but they didn't show me anything that I hadn't already tried.

Since posting this I did actually manage to get one of the agents to connect in the AMT only group after running the command multiple times so I know there is nothing wrong with the firewall configuration.

The AMT only group was just setup to see if I could get anything to work. I would like to get them working in the regular groups and I did set the policies up. I first tried the fully automatic option and then the other manual policies with CIRA enabled as well and even tried the deactivate option but still nothing would show up in the trace.

Now that I have managed to get one working in the AMT only group there is MPSCMD keepalive activity happening so at least I know the trace works as well.

Just to be clear on the steps I am currently trying:
1. I have Intel NUCs with 7th Gen i5 and vPro. AMT in it's completely stock unconfigured state and they are on a remote network with the agent installed connecting via the WAN.
2. I set the policy of the agent group they are in to Fully Automatic or CCM with Randomize Password, CIRA set to Connect to Server and I have tried both options for unknown password.
3. Nothing happens..
